cmd/swctl/cmd_manage.go

```diff
@@ -0,0 +1,637 @@
+	flagset.BoolVar(&opts.Force, "force", false, "Force the action")
+	flagset.StringVar(&opts.Target, "target", "", "Target config file used as base")
+	flagset.StringVar(&opts.Format, "format", "", "Format for the output (yaml, json, proto, go template..)")
+	flagset.BoolVar(&opts.DryRun, "dryrun", false, "Run without actually modifying anything")
+	flagset.BoolVar(&opts.ShowConfig, "show-config", false, "Print config for entity detail")
+	flagset.BoolVarP(&opts.Interactive, "interactive", "i", false, "Enable interactive mode")
+	flagset.StringToStringVar(&opts.Vars, "var", nil, "Override values for variables (--var VAR_NAME=value)")
+}
+
+func NewManageCmd(cli Cli) *cobra.Command {
+	var opts ManageOptions
+	cmd := &cobra.Command{
+		Use:              "manage ENTITY [ACTION]",
+		Short:            "Manage config with entities",
+		Long:             "Manages initial config setup and updates to config using entities",
+		Example:          color.Sprint(exampleManageCmd),
+		Args:             cobra.ArbitraryArgs,
+		TraverseChildren: true,
+		//DisableFlagParsing: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(cli.Entities()) == 0 {
+				return fmt.Errorf("no entities defined, use --entityfile=FILE to specify file to load from or add 'entities.yaml' to work directory")
+			}
+			return runManageCmd(cli, opts, args)
+		},
+	}
+	opts.InstallFlags(cmd.PersistentFlags())
+	return cmd
+}
+
+func runManageCmd(cli Cli, opts ManageOptions, args []string) error {
+	if len(args) > 2 {
+		return fmt.Errorf("expected at most 2 arguments (ENTITY ACTION), got %d arguments: %q", len(args), args)
+	}
+
+	// list all entities
+	if len(args) == 0 {
+		color.Fprintf(cli.Out(), "Listing <bold>%d</> loaded entities:\n\n", len(cli.Entities()))
+		for _, e := range cli.Entities() {
+			name := e.Name
+			desc := ""
+			if e.Description != "" {
+				desc = prefixTmpl(e.Description, "   ") + "\n"
+			}
+			color.Fprintf(cli.Out(), " - %s <gray>(%v vars)</>\n%s",
+				color.Yellow.Sprint(name), len(e.GetVars()), desc)
+		}
+		return nil
+	}
+
+	logrus.Tracef("running manage with args: %q %+v", args, opts)
+
+	// lookup entity
+	entityName := strings.ToLower(args[0])
+	var entity Entity
+	for _, e := range cli.Entities() {
+		if strings.ToLower(e.GetName()) == entityName {
+			entity = e
+			break
+		}
+	}
+	if entity.Name == "" {
+		return fmt.Errorf("entity %q not found", entityName)
+	}
+
+	logrus.Debugf("managing entity: %v (%d vars), config has %d bytes", entity.GetName(), len(entity.GetVars()), len(entity.Config))
+
+	// print entity detail
+	if len(args) == 1 {
+		color.Fprintf(cli.Out(), "<lightWhite>ENTITY</>:      %s\n", color.Yellow.Sprint(entity.Name))
+		if len(entity.Description) > 0 {
+			desc := strings.TrimSpace(entity.Description)
+			color.Fprintf(cli.Out(), "<lightWhite>DESCRIPTION</>: %s\n", color.FgDefault.Sprint(desc))
+		}
+		color.Fprintf(cli.Out(), "<lightWhite>ORIGIN</>: %s\n", color.Gray.Sprint(entity.Origin))
```

Comment:
Not sure how important the entity origin information is for the user, but the gray color formatting makes it a bit harder to see in dark-themed terminals. I think using gray is fine for the `press ENTER to confirm` and similar less important auxiliary output. If the entity origin falls into this category then it can be stay like this, else it could use a bit brighter color.